retrieval/target.go

```diff
@@ -304,7 +304,7 @@ func populateLabels(lset model.LabelSet, cfg *config.ScrapeConfig) (res, orig mo
 		}
 	}
 
-	preRelabelLabels := lset
+	preRelabelLabels := lset.Clone()
```

Comment:
I see that we modify the label set afterwards to remove the meta labels. I guess for UI purposes we'll always have to carry around the two sets (with and without meta labels).